mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -102,6 +107,24 @@ private[feature] trait Word2VecBase extends Params
     "be divided into chunks up to the size (> 0)", ParamValidators.gt(0))
   setDefault(maxSentenceLength -> 1000)
 
+  /**
+   * Choose the technique to use for generating word embeddings
+   * Default: true (Use Skip-Gram with Hierarchical softmax)
+   * @group param
+   */
+  final val skipGram = new BooleanParam(this, "skipGram", "Use Skip-Gram model to generate word " +
```

Comment:
How about making this a ``Param[String]``, ala [LDA](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala#L175)? Maybe named ``algorithm`` or something like that. That would make it clearer, and also make it easier to add any possible future algorithms (such as GloVe)